### PR TITLE
add scratch-mine script to print a users instances

### PR DESCRIPTION
**Pull request by umanwizard**
For example:

```
brennan@New-Orleans ~/c/materialize ❯❯❯ bin/scratch-mine
+-----------------------+---------------------+-------------------+-------------------------+---------------------+
|          Name         |     Instance ID     | Public IP Address |       Launched By       |     Delete After    |
+-----------------------+---------------------+-------------------+-------------------------+---------------------+
|    61043ea7-chbench   | i-04b3a98a9c33b6eba |   52.15.232.156   | brennan@materialize.com | 2021-08-17 09:39:55 |
|    f655656f-chbench   | i-0e5f601f257d90618 |    3.143.251.99   | brennan@materialize.com | 2021-08-17 09:53:46 |
|     f655656f-chaos    | i-0a5c4b54433649b2e |    3.22.166.191   | brennan@materialize.com | 2021-08-17 09:53:46 |
| 61043ea7-billing-demo | i-09e127d4390888e2a |   18.218.168.229  | brennan@materialize.com | 2021-08-17 09:39:55 |
| f655656f-billing-demo | i-05840f229cf135194 |   3.143.228.110   | brennan@materialize.com | 2021-08-17 09:53:46 |
|    f655656f-kinesis   | i-003bcc1ff5fb6924f |    3.133.82.205   | brennan@materialize.com | 2021-08-17 09:53:46 |
+-----------------------+---------------------+-------------------+-------------------------+---------------------+
```

```
brennan@New-Orleans ~/c/materialize ❯❯❯ bin/scratch-mine --all
+-------------------------+---------------------+-------------------+-------------------------+---------------------+
|           Name          |     Instance ID     | Public IP Address |       Launched By       |     Delete After    |
+-------------------------+---------------------+-------------------+-------------------------+---------------------+
| scratch kubernetes node | i-0c2f2cc88bc027a26 |        None       |           None          |         None        |
| scratch kubernetes node | i-0505cc41f2e64dbeb |        None       |           None          |         None        |
| scratch kubernetes node | i-0c521682e47efceb1 |        None       |           None          |         None        |
|     61043ea7-chbench    | i-04b3a98a9c33b6eba |   52.15.232.156   | brennan@materialize.com | 2021-08-17 09:39:55 |
|     f655656f-chbench    | i-0e5f601f257d90618 |    3.143.251.99   | brennan@materialize.com | 2021-08-17 09:53:46 |
|      f655656f-chaos     | i-0a5c4b54433649b2e |    3.22.166.191   | brennan@materialize.com | 2021-08-17 09:53:46 |
|  61043ea7-billing-demo  | i-09e127d4390888e2a |   18.218.168.229  | brennan@materialize.com | 2021-08-17 09:39:55 |
|  f655656f-billing-demo  | i-05840f229cf135194 |   3.143.228.110   | brennan@materialize.com | 2021-08-17 09:53:46 |
|     f655656f-kinesis    | i-003bcc1ff5fb6924f |    3.133.82.205   | brennan@materialize.com | 2021-08-17 09:53:46 |
+-------------------------+---------------------+-------------------+-------------------------+---------------------+
```

### CODE REVIEW

**Comment by umanwizard** [comment]
Note (for those of you who look at email), the correct title of this PR is "add scratch-mine script to print a users instances"

**Review by benesch — COMMENTED** [review]
Having multiple separate commands here seems a bit weird: can we design it so it's like:

```
bin/scratch create
bin/scratch mine
```

**Comment by umanwizard** [comment]
Sure -- I was thinking of doing so as a follow-up, but if you want me to do so as part of this PR, I can.

**Comment by benesch** [comment]
> Sure -- I was thinking of doing so as a follow-up, but if you want me to do so as part of this PR, I can.

Yeah, that'd be great. These things have a way of proliferating otherwise.

**Comment by umanwizard** [comment]
@benesch if you end up taking another look, you might have an idea how to eliminate the `# type: ignore` here.

File: bin/scratch [hunk 1]
```diff
@@ -9,6 +9,6 @@
 # the Business Source License, use of this software will be governed
 # by the Apache License, Version 2.0.
 #
-# mzcompose — runs Docker Compose with Materialize customizations.
+# scratch -- manipulate scratch resources
```
Comment (benesch) [inline bin/scratch:12]: hey, that's not an emdash!

**Review by benesch — COMMENTED** [review]
Yeah I'd probably avoid getting too clever with module structure and just tolerate a bit of duplication:

```py
import argparse

from materialize.cli.scratch import create, mine
from materialize import errors

def main() -> None:
    parser = argparse.ArgumentParser()
    subparsers = parser.add_subparsers()
    for name, configure, run in [
        ("create", create.configure_parser, create.run),
        ("mine", mine.configure_parser, mine.run),
    ]:
        s = subparsers.add_parser(name)
        configure(s)
        s.set_defaults(func=run)

    args = parser.parse_args()
    # TODO - Pass `required=True` to parser.add_subparsers once we support 3.7
    if "func" not in args:
        raise errors.BadSpec("Must specify a command")

    args.func(args)


if __name__ == "__main__":
    main()

```

**Review by benesch — APPROVED** [review]
❤️

**Pull request merged**